_posts/news/2023-01-26-announcing-scalajs-1.13.0.md

```diff
@@ -0,0 +1,155 @@
+
+This is a **minor** release:
+
+* It is backward binary compatible with all earlier versions in the 1.x series: libraries compiled with 1.0.x through 1.12.x can be used with 1.13.0 without change.
+* It is *not* forward binary compatible with 1.12.x: libraries compiled with 1.13.0 cannot be used with 1.12.x or earlier.
+* It is *not* entirely backward source compatible: it is not guaranteed that a codebase will compile *as is* when upgrading from 1.12.x (in particular in the presence of `-Xfatal-warnings`).
+
+As a reminder, libraries compiled with 0.6.x cannot be used with Scala.js 1.x; they must be republished with 1.x first.
+
+## Breaking changes
+
+### Drop support for Scala 2.11 and 2.12.1
+
+5 years after the end-of-life of Scala 2.11, Scala.js 1.13.0 drops its support.
+Support for Scala 2.11 has slowly but steadily increased in cost over the year, and we arrived at the issue that broke the proverbial camel's back.
+
+We announced a [request for comments issue](https://github.com/scala-js/scala-js/issues/4759) through all our usual communication channels.
+It gathered by far the highest number of upvotes any Scala.js issue ever received.
+
+Along with it, we dropped support for Scala 2.12.1, as it then remained the only version that does not support trailing commas.
+
+## Enhancements with compatibility concerns
+
+### Checked exceptions for `NullPointerException`
+
+As documented in [the semantics of Scala.js]({{ BASE_PATH }}/doc/semantics.html#undefined-behaviors), `NullPointerExceptions`s are Undefined Behavior in Scala.js, similarly to `ClassCastException`s and `ArrayIndexOutOfBoundsException`s.
+Since Scala.js 1.12.0 added checks for `ArrayStoreException`s and `NegativeArraySizeException`s, the `NullPointerException`s were the only remaining language-mandated exceptions that did not receive checks.
+
+Scala.js 1.13.0 fills that gap, and now checks `NullPointerException`s in development (fastLink) mode.
+As other undefined behavior errors, they will be reported as `UndefinedBehaviorError`s in fastLink mode, and unchecked in fullLink mode.
+
+These additional checks can and will slow down fastLink-generated code.
+Benchmarks show slowdowns of anywhere between 0% and 17%.
+As a reminder, we expect production code to be generated with `fullLinkJS`.
+fullLink-generated code is not affected by the checks and therefore not subject to this performance hit.
+
+If the performance hit in `fastLinkJS` is too much to bear in your particular case, you can disable the checks even in fastLink with the following sbt setting:
+
+{% highlight scala %}
+scalaJSLinkerConfig ~= {
+  import org.scalajs.linker.interface.CheckedBehavior
+  _.withSemantics(_
+    .withNullPointers(CheckedBehavior.Unchecked)
+  )
+}
+{% endhighlight %}
+
+In some rare situations, this may turn code that appeared to work into actively throwing an exception.
```

Comment:
This sentence is very misleading now that it is after the config block. Have you considered moving it up.

As other undefined behavior errors, they will be reported as `UndefinedBehaviorError`s in fastLink mode, and unchecked in fullLink mode.
In some rare situations, this may turn code that appeared to work into actively throwing an exception.

Further, these additional checks can and will slow down fastLink-generated code.
Benchmarks show slowdowns of anywhere between 0% and 17%.

Reply:
Ah yes indeed, ouch. I amended it as you suggested.